### PR TITLE
completed the catalog task

### PR DESCRIPTION
**Pull request by Shevtsooov**
- [DEMO LINK](https://Shevtsooov.github.io/layout_catalog/)
- [TEST REPORT LINK](https://Shevtsooov.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/styles/utils/functions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@function dynamicMaxWidth($width, $margins, $amount) {
+  @return $width * $amount + $margins * 3;
+}
```
Comment (snuber-w) [inline src/styles/utils/functions.scss:3]: nice work 🔥

File: src/styles/blocks/catalog.scss [hunk 2]
```diff
@@ -0,0 +1,129 @@
+@import "../utils/functions.scss";
```
Comment (snuber-w) [inline src/styles/blocks/catalog.scss:1]: Suggestion: it might be better to connect all files to main.scss 🤓

**Review by snuber-w — APPROVED** [review]
Great 🥇

File: src/styles/blocks/catalog.scss (hunk 2 above)
Comment (Shevtsooov) [reply src/styles/blocks/catalog.scss]: @snuber-w Thanks, done.

_I accidentally resolved the conversation_

**Review by silchyn — CHANGES_REQUESTED** [review]
> there should always be 4 cards in a row (not 2, 3 or 5)

**Comment by Shevtsooov** [comment]
@silchyn Hi, could you elaborate on the comment, please?
There are always 4 cards in a row starting with the width of about 950px.

**Comment by Shevtsooov** [comment]
@silchyn I see, seems like it should be still 4 in row when we redeuce the width.
Working on it.

**Review by niaYaro — APPROVED** [review]
Great job! ⭐

File: src/styles/blocks/catalog.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+.catalog {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 50px 0;
+  width: dynamicMaxWidth($card-width, $card-gap, $card-in-row);
+  min-width: 1024px;
+}
+
+.card {
```
Comment (silchyn) [inline src/styles/blocks/catalog.scss:14]: `container` and `card` are not part of `catalog` block

File: src/styles/blocks/catalog.scss [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+  &__buy-button:hover {
+    color: $button-bg-color;
+    background: white;
+    border-color: $button-bg-color;
+  }
+
+}
+
+.card--gap {
```
Comment (silchyn) [inline src/styles/blocks/catalog.scss:114]:
```suggestion
  &--gap {
```

File: src/styles/blocks/header.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+  &__logo {
+    display: block;
+    width: 40px;
+    height: 40px;
+    margin: 10px 0;
+  }
+}
+
+.nav {
```
Comment (silchyn) [inline src/styles/blocks/header.scss:17]: `nav` is not part of `header` block

File: src/styles/utils/functions.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+@function dynamicMaxWidth($width, $margins, $amount) {
+  @return $width * $amount + $margins * 3;
```
Comment (silchyn) [inline src/styles/utils/functions.scss:2]:
```suggestion
  @return $width * $amount + $margins * ($amount - 1);
```

**Review by silchyn — CHANGES_REQUESTED** [review]

**Comment by Shevtsooov** [comment]
@silchyn Thanks, all the mistakes were corrected.

File: src/styles/blocks/card.scss [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    text-decoration: none;
+    text-transform: uppercase;
+    color: $button-text-color;
+
+    background: $button-bg-color;
+    border: 1px solid transparent;
+    border-radius: 5px;
+  }
+
+  &__buy-button:hover {
```
Comment (silchyn) [inline src/styles/blocks/card.scss:93]:
```suggestion
    &:hover {
```

File: src/styles/blocks/card.scss [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    background: white;
+    border-color: $button-bg-color;
+  }
+
+  &--gap {
+    margin-right: $card-gap;
+    margin-bottom: 46px;
+  }
+  
+  &--gap:nth-child(4n) {
```
Comment (silchyn) [inline src/styles/blocks/card.scss:104]:
```suggestion
    &:nth-child(4n) {
```

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**